### ISSUE TITLE
Strange tags in DAWs

### ISSUE THREAD

**Issue by sub26nico** [sadko4u/lsp-plugins#44]
I notice that LSP plugins are classified in curious ways by many Digital Audio Workstation like Ardour or Reaper or host (Jalv.select). 
 - In Ardour, Parametric Equalizer plugins (lv2) are in 'Analyser' category and not in 'Equaliser'.
If 'analyser' button in the Ardour's plug-in manager window is not 'ON',  Parametric Equalizers are not found by name's search (lv2).
All LSP gates, Expanders and Limiters are in 'Dynamic' but there are 'Gate', 'Expander' and 'Limiter' categories. Impulse Reverb and Response plugins are tagged as 'Converter' (and not 'Reverb'), 
 All Ladspa are tagged 'unknow' and LinuxVST 'LXVST' (maybe Ardour's devs choices). 

- in Reaper, all Compressors, Limiters, Gates, Expanders and Dynamic Processor are in 'Mastering', Delay Compensators and Slapback Delays in 'Room'. 

- in jalv.select, Gates, Impulse Reverbs , Expanders and Limiters are in 'Dynamic' category, but there are 'gate', 'reverb', 'expander' and 'limiter' categories.

**Comment by sadko4u** [comment]
This is the result of understanding plugin classification for different plugin formats.

First of all, for LADSPA, there is **no** plugin classification.

For VST 2.x, there is a very poor plugin classification out of the box:
https://github.com/sadko4u/lsp-plugins/blob/04cc45a4aee3bd3a46ecbd0a9160d0d87f8f8a32/include/3rdparty/steinberg/vst2.h#L1522-L1588

By the other side, LV2 provides very rich and hierarchical plugin categorization:
http://lv2plug.in/ns/lv2core/lv2core.html#Plugin

And LSP plugins do reproduce the inheritance hierarchy in TTL. For example, for Gate plugin:
```
lsp:gate_mono
        a lv2:Plugin, doap:Project, lv2:DynamicsPlugin, lv2:GatePlugin ;
```

So actually if i right understand, hosts should locate this plugin into both categories: **Dynamics** and **Gates**.

Internally LSP plugins have the LV2-matching classification, for example the same gate:
https://github.com/sadko4u/lsp-plugins/blob/04cc45a4aee3bd3a46ecbd0a9160d0d87f8f8a32/src/metadata/gate.cpp#L17

where class constants are defined in the following way:
https://github.com/sadko4u/lsp-plugins/blob/04cc45a4aee3bd3a46ecbd0a9160d0d87f8f8a32/include/metadata/metadata.h#L155-L194

For VST with it's poor plugin categorizations the following code was written to decode plugin class:
https://github.com/sadko4u/lsp-plugins/blob/04cc45a4aee3bd3a46ecbd0a9160d0d87f8f8a32/src/container/vst.cpp#L217-L307

If you have any ideas how to improve the categorization, you're welcome for discussion.

**Comment by sadko4u** [comment]
@falkTX, @x42, @drobilla
Can you please clarify classification of LV2 and VST plugins from the host-side?
Should LSP plugins do something for better plugin classification?

**Comment by sub26nico** [comment]
> If you have any ideas how to improve the categorization, you're welcome for discussion.
@sadko4u , unfortunately, I haven't any coding skills. All I can do is making bug reports and testing. 
I'ld be happy to test if any changes are proposed.

**Comment by x42** [comment]
> In Ardour, Parametric Equalizer plugins (lv2) are in 'Analyser' category

The problem is that LSP plugins define *multiple* categories.

```
a lv2:Plugin, doap:Project, lv2:FilterPlugin, lv2:EQPlugin, lv2:ParaEQPlugin, lv2:AnalyserPlugin
```
The last one wins, which happens to be Analyzer.
Unlike tags, categories are organized in a tree and you can pick at most one.

If you look at http://lv2plug.in/ns/lv2core/lv2core.html : ParaEQPlugin is a subclass of  EQPlugin, which is a subclass of FilterPlugin which is a subclass of Plugin. 
You can safely remove ` lv2:FilterPlugin, lv2:EQPlugin` from the .ttl - all those are implicit.

**Comment by sadko4u** [comment]
@x42 thanks for clarification. That's sad that we can define multiple classes for plugin but the host won't properly handle them.

Anyway, I've created **github-issue-44** branch and pushed the commit which corrects the plugin metadata: 64ed9f558624bcce78d56c18cfadbe69e7f0be9d
The only thing is left: I can not find category for Trigger plugin. So actually now it is a 'dynamics' plugin.

@sub26nico is it possible for you to build from this branch and test the categorization?

**Comment by drobilla** [comment]
> The last one wins, which happens to be Analyzer.

This isn't what is happening here, there is no "last" one, the order of statements in the file is irrelevant since this is loaded into a model.  `lilv_plugin_get_class()` selects the _first_ URI it finds, with is Analyzer in this case because it is first alphabetically.

That said, hosts are not limited to this, so if multiple categories make sense, it's fine to use them.  You shouldn't use redundant classes that are parents of other classes, though.

In this case I don't think Analyzer is an appropriate category for EQs.

**Comment by sub26nico** [comment]
@sadko4u , i try to build it and test the categorization today.

**Comment by x42** [comment]
For future reference, Ardour6 will use user-editable tags, the category is only used to pre-seed the list. The filter-buttons (Analyzer, ..) are gone and there's prospective search using name and tag.

While on the subject: The long names of the plugins make it hard to identify them on the mixer-strip.

Since they're all "english name - german name" did you consider using .ttl language tags for them
` doap:name "English Name" , "German Name"@de ;` ?
see also http://lv2plug.in/book/#_amp_ttl

**Comment by sub26nico** [comment]
@sadko4u , i built it and there are lot of improvements. 
Parametric Equalizer are now in 'Parametric', Graphic Equalizer in 'Multiband', all others except triggers are OK. For triggers, other ones on my system  are in 'Converter' or 'Utility'.

**Comment by sadko4u** [comment]
Some minor update according @x42 's recommendations: 53ba0deda24b8b929f663c4d353aa03f4928273d
Added 'LSP' prefix to LV2 plugin names for better distinguishing from other plugins.

**Comment by x42** [comment]
Isn't that counter-productive? I suggested *shorter*, more concise names :)

Guess what the following ellipsized name stands for: ![guess_what](https://user-images.githubusercontent.com/30246/55196114-b5eedb80-51ae-11e9-99c2-a379973059f6.png) :)

In any case LSP prefix isn't needed, you can filter by Author just fine:
![plugin_manager_lsp_filter](https://user-images.githubusercontent.com/30246/55196103-acfe0a00-51ae-11e9-9bac-ebe5292ee766.png)

**Comment by sadko4u** [comment]
> Isn't that counter-productive? I suggested _shorter_, more concise names :)
In current circumstances, it isn't. Consider I'm a user and want to find proper reverb plugin. What I will see if I will navigate through categories? Let's examine 'Reverb' category:
![изображение](https://user-images.githubusercontent.com/12834464/55198732-4685e580-51c8-11e9-810d-33c2ca2c21ab.png)
Oh, I see 'Calf Reverb', 'dRow Audio Reverb', 'TAL Reverb' and some unknown 'Impulse Responses Mono' from unknown provider (consider you don't see 'LSP' prefix on this screenshot). Hmm, will I try them first or will leave them gather a dust?

It could be reasonable if the author/provider was visible near the plugin name. But now we see only plugin format which is, obviously, LV2 because all other plugin types are not categorized (at least, under Linux).

Since some plugins embed their bundle/brand name into the plugin name, other plugins become discriminated, don't you think so?

**Comment by x42** [comment]
fair enough

**Comment by sub26nico** [comment]
@sadko4u I've build with the last commit and I can confirm searching plugins is easier with LSP in the name. And for the name in mixer (screenshot by @x42 ) , it only happens when the mixer track have the smaller size.

**Comment by sadko4u** [comment]
Merged changes into **devel** branch: 67f159f5fa5c4e095de74255107f4ab68025d6f5
The change will be published in the nearest release.

**Comment by brummer10** [comment]
> Isn't that counter-productive? I suggested _shorter_, more concise names :)

While a search prefix is nice, I would second to shorten the plugin names, it shouldn't be a description or, a translation included. This ultra long names blow up the search fields unnecessary, or needs to be truncate.

**Comment by sadko4u** [comment]
> While a search prefix is nice, I would second to shorten the plugin names, it shouldn't be a description or, a translation included. This ultra long names blow up the search fields unnecessary, or needs to be truncate.

You're pretty right but it's always a hard to find compromise between well-known long names and short cryptic names.
Consider I'll shorten 'Parametric Equalizer x16 Mid/Side' to just 'Parametric Equalizer Mid/Side'. But there is also plugin with 32 independent filters. How I will distinguish equalizers?
Alright, I'll keep the 'Parametric Equalizer x16'. But there what type of equalizer will I take - Mono, Stereo, Mid/Side or Left/Right? How can I distinguish these instances?
OK, then what if I remove 'Parametric' from name? But there is also a 'Graphic' version of equalizer.
I think there is no reason to remove 'Equalizer' from the name of plugin, too.
The only thing left is to keep the model name - PE16LR. But if you've just installed plugin package, how you will decrypt this acronym without reading documentation?

If you have better idea how to deal with it, you can suggest your alternatives and we can discuss them.

**Comment by brummer10** [comment]
What I do now is truncate the names at " - " 
The resulting names been meaningful enough I think, and my search/ list field could hold them without been expanded. So, for example:
`Sidechain Dynamic Processor LeftRight - Sidechain-Dynamikprozessor LeftRight`
becomes:
`Sidechain Dynamic Processor LeftRight`
which is as meaningful as the above name.

**Comment by sadko4u** [comment]
> What I do now is truncate the names at " - "
> The resulting names been meaningful enough I think, and my search/ list field could hold them without been expanded. So, for example:

Alright, it's clean now what you mean. In the 1.1.9 release this case is actual.
I've checked the **devel** branch and it seems that plugins names are showing as you've said.
The changes will appear in 1.1.10 release which is currently delayed due to huge amount of work I need to do before publishing it.

**Comment by brummer10** [comment]
Great.

**Comment by sadko4u** [comment]
Available in 1.1.10 release!
